client/my-sites/site-settings/utils.ts

```diff
@@ -13,9 +20,48 @@ type Site = {
 	plan: Plan;
 };
 
-export function hasSiteAnalyticsFeature( site: Site ): boolean | undefined {
+/**
+ * Returns true if the site is on a free plan and a WPCOM site (Simple or Atomic).
+ *
+ * @param site The site to check.
+ * @param state Global state tree.
+ * @param siteId Site ID to check.
+ * @returns bool True if the site is on a free pln and a WPCOM site, false otherwise.
+ */
+export function isFreeWPCOMSite( site: Site, state: string, siteId: string ): boolean | null {
```

Comment:
Now this is using state, it's essentially a selector. Should we move these helpers to live with the other selectors?

Reply:
I moved / converted them to selectors: https://github.com/Automattic/wp-calypso/pull/53196/commits/0e63e216afb7d078af1935ed40c02364ae1140b7

I'm getting this warning though: "Do not pass complex objects as arguments for a memoized selector". Maybe `site` is too complex. 🤔 

I should probably also add selector tests to replace the util tests I deleted.